src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let cloneState = { ...state };
+
+  for (const action of actions) {
```

Comment:
You don't need to write `action` every time to get object property, just destructurize them right away
```suggestion
  for (const { type, extraData .... } of actions) {
```